src/server/pegasus_hotkey_collector.cpp

```diff
@@ -0,0 +1,279 @@
+// Copyright (c) 2017, Xiaomi, Inc.  All rights reserved.
+// This source code is licensed under the Apache License Version 2.0, which
+// can be found in the LICENSE file in the root directory of this source tree.
+
+#include "pegasus_hotkey_collector.h"
+
+#include "base/pegasus_key_schema.h"
+#include "base/pegasus_rpc_types.h"
+#include <math.h>
+
+namespace pegasus {
+namespace server {
+
+hotkey_collector::hotkey_collector(dsn::apps::hotkey_type::type hotkey_type,
+                                   const std::string &replica_name)
+    : hotkey_type(hotkey_type),
+      _collector_state(STOP),
+      _coarse_result(-1),
+      _replica_name(replica_name)
+{
+    hotkey_collector_max_work_time = dsn_config_get_value_uint64(
+        "pegasus.server",
+        "hotkey_collector_max_work_time",
+        150,
+        "hotkey_collector_max_work_time, after that the collection will stop automatically");
+
+    coarse_capture_hash_bucket_num =
+        dsn_config_get_value_uint64("pegasus.server",
+                                    "coarse_capture_hash_bucket_num",
+                                    37,
+                                    "the number of coarse capture hash buckets");
+
+    data_variance_threshold =
+        dsn_config_get_value_uint64("pegasus.server",
+                                    "data_variance_threshold",
+                                    3,
+                                    "the threshold of variance_cal to find the outliers");
+
+    _collector_start_time = dsn_now_s();
+
+    hotbucket_hash = [](const std::string &data, const int split_count) {
+        return (int)(std::hash<std::string>{}(data) % split_count);
+    };
+}
+
+bool hotkey_collector::init()
+{
+    switch (_collector_state.load()) {
+    case COARSE:
+        ddebug("Now is detecting %s hotkey, state is %s",
+               hotkey_type == dsn::apps::hotkey_type::READ ? "read" : "write",
+               "COARSE");
```

Comment:
Use get_status to simplify this function

Reply:
How to simplify?